File: example/examples/overlays/polygon.js
import React, { Component } from 'react'
import { StyleSheet } from 'react-native'
import { MapView } from 'react-native-baidumap-sdk'

const points = [
  {
    latitude: 39.806901,
    longitude: 116.397972,
  },
  {
    latitude: 39.806901,
    longitude: 116.297972,
  },
  {
    latitude: 39.906901,
    longitude: 116.397972,
  },
]

export default class Polygon extends Component {
  static navigationOptions = { title: 'Polygon' }

  render() {
    return (
      <MapView style={StyleSheet.absoluteFill}>
        <MapView.Polygon
          points={points}
          strokeWidth={2}
          strokeColor="rgba(0, 0, 255, 0.5)"
          fillColor="rgba(255, 0, 0, 0.5)"
        />
      </MapView>
    )
  }
}

File: lib/android/src/main/java/qiuxiang/baidu_map/map_view/Polygon.kt
package qiuxiang.baidu_map.map_view

import android.content.Context
import android.graphics.Color
import com.baidu.mapapi.map.BaiduMap
import com.baidu.mapapi.map.Polygon
import com.baidu.mapapi.map.PolygonOptions
import com.baidu.mapapi.map.Stroke
import com.baidu.mapapi.model.LatLng
import com.facebook.react.views.view.ReactViewGroup

class Polygon(context: Context) : ReactViewGroup(context), Overlay {
  private var polygon: Polygon? = null

  var points: List<LatLng> = emptyList()
    set(value) {
      field = value
      polygon?.points = value
    }

  var strokeWidth: Int = 1
    set(value) {
      field = value
      polygon?.stroke = Stroke(value, strokeColor)
    }

  var strokeColor: Int = Color.BLACK
    set(value) {
      field = value
      polygon?.stroke = Stroke(strokeWidth, value)
    }

  var fillColor: Int = Color.BLACK
    set(value) {
      field = value
      polygon?.fillColor = value
    }

  var zIndex: Int = 0
    set(value) {
      field = value
      polygon?.zIndex = value
    }

  override fun add(map: BaiduMap) {
    polygon = map.addOverlay(
      PolygonOptions()
        .points(points)
        .stroke(Stroke(strokeWidth, strokeColor))
        .fillColor(fillColor)
        .zIndex(zIndex)
    ) as Polygon
  }

  override fun remove() {
    polygon?.remove()
  }
}
